src/oauth-facade/index.ts

```diff
@@ -30,7 +31,7 @@ setFlag('IS_LOG', false);
  * This wraps some messy details of low-level library usage.
  */
 export default class OauthFacade {
-  protected readonly config: AuthorizationServiceConfiguration;
```

Comment:
I realize that we can't have async constructors, but it's bad that we have to do this instead.

The original goal of `OauthFacade` was to abstract over a bunch of weirdness in how `@openid/appauth` needed to be delicately constructed. I'm actually fine to say that "endpoint discovery is outside the scope of `@openid/appauth` and therefore outside the scope of `OauthFacade`". If we do that, then the constructor parameters to `OauthFacade` should actually be the fully-formed `authorizationEndpoint`, `tokenEndpoint`, `revocationEndpoint` (and maybe we don't even need `authorizationServerBaseUrl` anymore).

We could then push the async endpoint discovery work up to the `OauthClient` or into a static class method of `OauthFacade` (and construct it with a factory pattern).

Reply:
However, I realize that since `OauthClient` declares `protected readonly oauthFacade: OauthFacade;`, then we're really just pushing the inability to synchronously initialize things up a level.

This still might be a cleaner internal archecture overall, but I realize no approach is ideal.